anvil/src/eth/api.rs

```diff
@@ -0,0 +1,1018 @@
+        for n in lowest..highest + 1 {
+            // <https://eips.ethereum.org/EIPS/eip-1559>
+            if let Some(block) = fee_history.get(&n) {
+                response.base_fee_per_gas.push(U256::from(block.base_fee));
+                response.gas_used_ratio.push(block.gas_used_ratio);
+
+                // requested percentiles
+                if !reward_percentiles.is_empty() {
+                    let mut block_rewards = Vec::new();
+                    let resolution_per_percentile: f64 = 2.0;
+                    for p in &reward_percentiles {
+                        let p = p.clamp(0.0, 100.0);
+                        let index = ((p.round() / 2f64) * 2f64) * resolution_per_percentile;
+                        let reward = if let Some(r) = block.rewards.get(index as usize) {
+                            U256::from(*r)
+                        } else {
+                            U256::zero()
+                        };
+                        block_rewards.push(reward);
+                    }
+                    rewards.push(block_rewards);
+                }
+            }
+        }
+
+        response.reward = Some(rewards);
+
+        // calculate next base fee
+        if let (Some(last_gas_used), Some(last_fee_per_gas)) =
+            (response.gas_used_ratio.last(), response.base_fee_per_gas.last())
+        {
+            let elasticity = self.backend.elasticity();
+            let last_fee_per_gas = last_fee_per_gas.as_u64() as f64;
+            if last_gas_used > &0.5 {
+                // increase base gas
+                let increase = ((last_gas_used - 0.5) * 2f64) * elasticity;
+                let new_base_fee = (last_fee_per_gas + (last_fee_per_gas * increase)) as u64;
+                response.base_fee_per_gas.push(U256::from(new_base_fee));
+            } else if last_gas_used < &0.5 {
+                // decrease gas
+                let increase = ((0.5 - last_gas_used) * 2f64) * elasticity;
+                let new_base_fee = (last_fee_per_gas - (last_fee_per_gas * increase)) as u64;
+                response.base_fee_per_gas.push(U256::from(new_base_fee));
+            } else {
+                // same base gas
+                response.base_fee_per_gas.push(U256::from(last_fee_per_gas as u64));
+            }
+        }
+
+        Ok(response)
+    }
+
+    /// Introduced in EIP-1159, a Geth-specific and simplified priority fee oracle.
+    /// Leverages the already existing fee history cache.
+    ///
+    /// Handler for ETH RPC call: `eth_maxPriorityFeePerGas`
+    pub fn max_priority_fee_per_gas(&self) -> Result<U256> {
+        Err(BlockchainError::RpcUnimplemented)
+    }
+
+    /// Returns traces for the transaction hash
+    ///
+    /// Handler for RPC call: `debug_traceTransaction`
+    pub async fn debug_trace_transaction(&self, _tx_hash: H256) -> Result<Vec<Trace>> {
+        Err(BlockchainError::RpcUnimplemented)
+    }
+}
+
+// == impl EthApi anvil endpoints ==
+
+impl EthApi {
```

Comment:
@brockelmore I've added several non-eth RPC handlers in this section, what's missing?